### DISCUSSION TITLE
Is there recombination collision type included?

### DISCUSSION THREAD

**Discussion by ZippoFight** [epochpic/epoch#453] [Q&A]
Hello, i want to model discharge process, where the plasma has a low temperature, with EPOCH. So I am wondering if there are collision types like excitaion and recombination in this code. Thank you in advance.

**Comment by Status-Mirror** [comment]
Hi @ZippoFight

Currently we do not have any form of recombination in the code, but we would be interested in adding it.

Recently we upgraded collisional ionisation and numerically reproduced [high-energy high-charge states of gold ions](https://www.nature.com/articles/s41598-022-23148-2), but found we were over-estimating the ion charge state. We believed the discrepancy between simulation and experiment was the fact we were lacking recombination of gold ions, so we have looked into this in more detail. Unfortunately, we were unable to find good models for any form of recombination for this system.

I appreciate this is very different to the recombination system you're describing. Are you aware of any theory to describe the recombination rates in your system?

Cheers,
Stuart

**Comment by Lingen1218** [comment]
@Status-Mirror May I ask what is the current status of recombination physics module in EPOCH?  Is it already developed so that the users could test it?

**Reply by DanRRRR** [reply]
Stuart, Where in this paper you find that you overestimate Z? 
With Gold people get Z~72 and estimate that 77 is also doable

**Comment by TomGoffrey** [comment]
Recombination will be included in the upcoming release, 4.20, as described in: https://doi.org/10.1103/PhysRevE.111.065209

If anyone is interested in using/testing it before then let me know and I can include instructions here.

**Reply by Haofan-sun** [reply]
Dear friends
Is the recombination module auto enabled in 4.20devel? or I need code to activate it.
If I need to activate it could you please tell me the code to activate recombination module.I can't find the code on user guide.
Best regards

**Reply by TomGoffrey** [reply]
By default it is disabled. You can enable it by setting `recombine = T` in species blocks. Note that this feature is experimental (hence it not being in an official release), and I'm still fixing issues with it. In particular the current implementation will fail if you enable recombination with thermal energies << ionisation energy of species. Any feedback is of course welcome and appreciated.

**Reply by Lingen1218** [reply]
Does it mean we have to set an initial high temperature to make ```recombination``` work? The thing is, we're trying to simulate a target that's tens of microns thick, and the deeper parts don’t really get heated by the laser.

**Reply by TomGoffrey** [reply]
Some of the models within the recombination package become exponentially large as the thermal energy becomes small compared to the ionisation energy. This can effectively make the recombination instantaneous.

If you are using the most recent version of the 4.20-devel branch there is an artificial limit placed on the three body recombination rate. This is introduced to prevent floating point issues. This should still yield an effective instantaneous recombination, but in a numerically safe way. However, this has only had limited testing, so please report any problems you encounter.

**Reply by Lingen1218** [reply]
Do you mean that the three-body recombination rate has a cutoff or maxima at low thermal energies? How about the other two recombination models: dielectronic and radiative recombination? We are currently testing it and will report any issues as soon as they arise.

**Reply by TomGoffrey** [reply]
The three-body rate scales as e^{ionisation_energy / kT), so in the limit of kT << ionisation_energy this can become (infinitely) large. Therefore we place an upper limit on the rate calculated, where the limit is calculated assuming kT = 0.01 * ionisation_energy. The remaining models don't have such a limit.

**Reply by Lingen1218** [reply]
We have run two simulations for a Cu target with/without recombination, and found that both cases give very similar charge states at 0.5 ps after the laser-solid interaction, which is unexpected. My question is: are the recombination rates for Cu included in the source code by default, or do users need to generate the atomic data themselves?

**Reply by TomGoffrey** [reply]
The recombination rates should be automatically calculated. What rates do you expect for the conditions at the end of the laser pulse?

**Reply by Lingen1218** [reply]
The temperature in the solid Cu target reaches ~5kev in the heating depth, that should give 3-body recombination rate of Cu22+ ~ 0.01/fs according to NRL formula. So I expect the charge state should be modified after 100 fs, which wewe didn't see in the test simulations. Is there a way to output the recombination rates from the simulations?

**Reply by TomGoffrey** [reply]
You can enable output of rates with the compiler flag `DTRANSITION_RATES,`. You can then enable `rate_3br`, `rate_rr` and `rate_dr` for the different recombination types. You can also output `rate_fi` and `rate_ci` for field and collision ionisation respectively. Would you be able to share the input.deck file you are using for testing?

**Reply by TomGoffrey** [reply]
Which NRL formula were you using? I can only see one for a singly ionised plasma, and that has a different temperature dependence than the model in EPOCH.

**Reply by Lingen1218** [reply]
The recombination rate was scaled with  the charge state  to Z^5.

**Reply by TomGoffrey** [reply]
The equation from the NRL (which in turn is taken from Zel'dovich and Raizer) appears to be only approximate. In particular I note (Zel'dovich and Raizer, bottom of page 407), it appears to require I >> kT. In general the dependence on the electron temperature and ionisation energy is a lot more complex. See equations 8.22, 8.45, 8.47 of Kremp's "Quantum statistics of non-ideal plasmas" which is the basis of the model used in EPOCH.

**Comment by Lingen1218** [comment]
In our test simulation, the 3-body recombination rate increases to an infinitely large value at ~ 30 fs after the peak of the laser pulse hits the front surface of the target. It seems that the upper limit was not working. Could you give me a hint on how to solve this issue?

<img width="1758" height="1010" alt="image" src="https://github.com/user-attachments/assets/4ccacf29-e3d3-4c3f-918d-8f7b60f181a6" />

**Reply by TomGoffrey** [reply]
Do you actually see overflow/infinite rates? This number is large, but still finite.

I can investigate further, but would need the input deck you are using please.

**Reply by Haofan-sun** [reply]
I do a simple simulation without laser.10um 48nc Cu+1 and electron with 1keV initial temperature.I run the same input under normal setting and with 3_body recombination disabled.The PIC results are quite different.
[input.txt](https://github.com/user-attachments/files/22022514/input.txt)
This is simulation without 3_body recombination.
[without3br.bmp](https://github.com/user-attachments/files/22022541/without3br.bmp)
This is simulation with 3_body recombination.
[with3br.bmp](https://github.com/user-attachments/files/22022554/with3br.bmp)
The figure at the top left shows the average charge state of Cu.
